Upgrade to JUnit 4.13.1. Fixes #827

nikhilnanivadekar opened the pull request:
I have broken the PR down in separate commits to make it easy for the review. Most of the changes are straightforward.

Please note, I have purposely not updated the copyright to make it easy for the review. We are almost in November now, and my plan is that in January I will update all copyright headers to the new year to make it easy for the contributors next year.

prathasirisha commented on eclipse-collections-forkjoin/src/test/java/org/eclipse/collections/impl/forkjoin/FJIterateTest.java:
```diff
@@ -265,7 +267,7 @@ public void testForEachImmutable()
     public void testForEachWithException()
     {
-        Verify.assertThrows(RuntimeException.class, () -> FJIterate.forEach(
+        assertThrows(RuntimeException.class, () -> FJIterate.forEach(
```
Suggestion to make this `Assert.assertThrows` instead of a static import, so it is consistent with the rest of the assertions in this class. Same comment for lines 331 and 495.

nikhilnanivadekar replied:
We have other such instances where we us static import. I dont understand the consistency aspect here. That said, I have fixed it.

prathasirisha reviewed (commented):
Noticed a file called version.properties has junit version (4.12) in it, wanted to bring that to your attention. Assuming that too needs to be updated.

donraab commented on eclipse-collections-testutils/src/test/java/org/eclipse/collections/impl/test/AssertNotEqualsThrowsTest.java:
```diff
@@ -38,7 +42,7 @@ public void objectObject()
         catch (AssertionError e)
         {
-            Verify.assertContains(AssertNotEqualsThrowsTest.class.getName(), e.getStackTrace()[0].toString());
+            Verify.assertContains("org.junit.Assert.fail", e.getStackTrace()[0].toString());
```
Make the string Static final.

nikhilnanivadekar replied:
How does that matter in this test? This test is deprecated as well.

donraab commented on eclipse-collections-testutils/src/test/java/org/eclipse/collections/impl/test/VerifyTest.java:
```diff
@@ -475,7 +475,7 @@ public void assertNotEqualsFailsOnSameReference()
         catch (AssertionError e)
         {
-            Verify.assertContains(VerifyTest.class.getName(), e.getStackTrace()[0].toString());
+            Verify.assertContains("org.junit.Assert.fail", e.getStackTrace()[0].toString());
```
Make the String static final.

nikhilnanivadekar replied:
How does that matter in a test?

donraab commented on unit-tests/unit-tests.iml:
```diff
@@ -11,12 +11,15 @@
     <orderEntry type="library" scope="TEST" name="Maven: org.hamcrest:hamcrest-core:1.3" level="project" />
-    <orderEntry type="library" scope="TEST" name="Maven: org.slf4j:slf4j-api:1.7.25" level="project" />
-    <orderEntry type="library" scope="TEST" name="Maven: org.slf4j:slf4j-nop:1.7.25" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: org.slf4j:slf4j-api:1.7.30" level="project" />
```
Why is this changed, but the ones above for slf4j are not?

nikhilnanivadekar replied:
will revert this

nikhilnanivadekar replied:
Reverted

donraab reviewed (changes requested):
This PR looks almost ready to go. I have a couple comments inline.

motlin commented on unit-tests-java8/src/test/java/org/eclipse/collections/test/IterableTestCase.java:
```diff
@@ -141,8 +141,7 @@ static void assertEquals(Object o1, Object o2)
             }
-        }
-        else if (o1 instanceof SortedMap<?, ?> || o2 instanceof SortedMap<?, ?>)
+        } else if (o1 instanceof SortedMap<?, ?> || o2 instanceof SortedMap<?, ?>)
```
Whitespace nit here

nikhilnanivadekar replied:
This was fixed in a subsequent commit.

nikhilnanivadekar commented:
@prathasirisha I have fixed the static import and removed the version.properties file as a separate commit. It was needed for ant builds which we dont do anymore.
@donraab I have reverted the iml file. I quite disagree with the static string comment, it is cumbersome in a test to infer such strings. I would prefer not to do that.

donraab reviewed (approved):
LGTM.
